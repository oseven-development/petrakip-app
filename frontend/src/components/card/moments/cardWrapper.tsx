import {
  IonCardSubtitle,
  IonCardTitle,
  IonCardContent,
  IonCardHeader,
  IonCard,
  IonRouterLink,
} from '@ionic/react'

export interface PropsCardWrapper {
  title: string
  subtitle?: string
  momentId: number
  children?: (false | JSX.Element)[]
}

export const CardWrapper = ({
  children,
  title,
  subtitle,
  momentId,
}: PropsCardWrapper) => (
  <IonCard>
    <IonCardHeader>
      <IonCardSubtitle>{subtitle}</IonCardSubtitle>
      <IonRouterLink routerLink={`/moments/details/${momentId}`}>
        <IonCardTitle>{title}</IonCardTitle>
      </IonRouterLink>
    </IonCardHeader>
    <IonCardContent>{children}</IonCardContent>
  </IonCard>
)
