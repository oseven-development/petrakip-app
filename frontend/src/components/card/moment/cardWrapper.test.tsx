import { render } from '@testing-library/react'

import { CardWrapper } from './cardWrapper'

describe('CardWrapper Component', () => {
  const myFalse: false = false
  const cardWrapper = (
    <CardWrapper title="Headline" momentId={123}>
      {myFalse}
      <h1>IAM A children</h1>
    </CardWrapper>
  )

  test('renders without crashing', () => {
    const { container } = render(cardWrapper)
    expect(container).toMatchSnapshot()
  })
})
