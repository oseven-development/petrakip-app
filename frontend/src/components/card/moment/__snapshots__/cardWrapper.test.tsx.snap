// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrapper Component renders without crashing 1`] = `
<div>
  <ion-card>
    <ion-card-header>
      <ion-card-subtitle />
      <ion-router-link
        href="/moments/details/123"
        router-link="/moments/details/123"
      >
        <ion-card-title>
          Headline
        </ion-card-title>
      </ion-router-link>
    </ion-card-header>
    <ion-card-content>
      <h1>
        IAM A children
      </h1>
    </ion-card-content>
  </ion-card>
</div>
`;
